File: AppWidgetDemo/app/src/main/java/com/example/akriti/appwidgetdemo/MainActivity.java
package com.example.akriti.appwidgetdemo;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (getIntent().getExtras() != null) {
            String selectedOption = getIntent().getExtras().getString(WidgetProvider.EXTRA_STRING);

            if (getIntent() != null && getIntent().getExtras() != null
                    && getIntent().getExtras().getString(WidgetProvider.EXTRA_STRING) != null) {
                Toast.makeText(this, "Opened " + selectedOption,
                        Toast.LENGTH_LONG).show();

                if (selectedOption != null) {
                    Intent intent = new Intent(MainActivity.this, ChampInfoActivity.class);
                    intent.putExtra(Constants.SELECTED_CHAMP, selectedOption);
                    startActivity(intent);
                }
            }
        }
    }
}

File: LifeCycleDemoApp/app/src/main/java/com/example/akriti/lifecycledemoapp/MainActivity.kt
package com.example.akriti.lifecycledemoapp

import android.content.Intent
import android.net.Uri
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.google).setOnClickListener {
            val uris = Uri.parse("https://www.google.com")
            val intents = Intent(Intent.ACTION_VIEW, uris)
            startActivity(intents)
        }

        findViewById<Button>(R.id.secondActivity).setOnClickListener {
            val intent = Intent(this@MainActivity, SecondActivity::class.java)
            startActivity(intent)
        }
    }

    override fun onStart() {
        super.onStart()
        Log.d("MainActivity", "Called on Activity Start")
    }

    override fun onResume() {
        super.onResume()
        Log.d("MainActivity", "Called on Activity Resume")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("MainActivity", "Called on Activity Destroy")
    }

    override fun onPause() {
        super.onPause()
        Log.d("MainActivity", "Called on Activity Pause")
    }

    override fun onStop() {
        super.onStop()
        Log.d("MainActivity", "Called on Activity Stop")
    }

    override fun onRestart() {
        super.onRestart()
        Log.d("MainActivity", "Called on Activity Stop")
    }
}
